### PR TITLE
groovysdk 3.0.8

### PR DESCRIPTION
**Pull request by samford**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

This updates `groovysdk` to 3.0.8 while also updating the `livecheck` block to check the download page, which links to the `stable` archive on Artifactory.

Incidentally, part of #74335.

### CODE REVIEW

**Review by nandahkrishna — APPROVED** [review]
I think we can merge this now and revisit bottling after Homebrew/brew#11253.

**Comment by samford** [comment]
It looks like #75946 will remove the `bottle :unneeded` line in `groovysdk`, so I'll just make sure to rebase that PR after this is merged 👍

**Pull request closed without merging**
